src/runtime/vdom/vdom-render.ts

```diff
@@ -985,26 +976,45 @@ render() {
         }
       }
 
-      for (i = 0; i < relocateNodes.length; i++) {
-        relocateData = relocateNodes[i];
-        nodeToRelocate = relocateData.$nodeToRelocate$;
+      for (const relocateData of relocateNodes) {
+        const nodeToRelocate = relocateData.$nodeToRelocate$;
         const slotRefNode = relocateData.$slotRefNode$;
 
         if (slotRefNode) {
-          // by default we're just going to insert it directly
-          // after the slot reference node
-          parentNodeRef = slotRefNode.parentNode;
-          insertBeforeNode = slotRefNode.nextSibling;
-          orgLocationNode = nodeToRelocate['s-ol'] as any;
-
-          while ((orgLocationNode = orgLocationNode.previousSibling as any)) {
-            refNode = orgLocationNode['s-nr'];
-            if (refNode && refNode['s-sn'] === nodeToRelocate['s-sn'] && parentNodeRef === refNode.parentNode) {
-              refNode = refNode.nextSibling as any;
-              if (!refNode || !refNode['s-nr']) {
-                insertBeforeNode = refNode;
-                break;
+          const parentNodeRef = slotRefNode.parentNode;
+          // When determining where to insert content, the most simple case would be
+          // to relocate the node immediately following the slot reference node. We do this
+          // by getting a reference to the node immediately following the slot reference node
+          // since we will use `insertBefore` to manipulate the DOM.
+          //
+          // If there is no node immediately following the slot reference node, then we will just
+          // end up appending the node as the last child of the parent.
+          let insertBeforeNode = slotRefNode.nextSibling as d.RenderNode | null;
```

Comment:
Do we need this type assertion? I'm not sure what purpose it's serving here ATM

Reply:
I saw https://github.com/ionic-team/stencil/pull/4994#discussion_r1373518603 after I approved this, but I'm not sure I understand (but also, my brain is actively turning to goo)

Reply:
We need to cast it since we may reassign this value later and the types won't match. `nextSibling` is of type `ChildNode | null | undefined`, but in the while loop we reassign a node that could be `RenderNode | null | undefined`. I tried to update our `RenderNode` interface to type `nextSibling` and `previousSibling` as `RenderNode | null`, but that broke types in some other places, so didn't wanna fight with that right now.